detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -77,9 +92,53 @@ private List<String> parseGav(String line) {
             }
         }
 
+        String projectName = metadata.getOrDefault(PROJECT_NAME_PREFIX, "orphanProject");
+        String rootProjectName = metadata.getOrDefault(ROOT_PROJECT_NAME_PREFIX, "");
+        String projectParent = metadata.getOrDefault(PROJECT_PARENT_PREFIX, "null");
+
+        addRelation(projectParent, projectName, rootProjectName);
+
+        if(gavPieces.size() == 3) {
+            String dependencyGroupName = gavPieces.get(0) + ":" + gavPieces.get(1);
+            if ((cleanedOutput.contains(STRICTLY) || cleanedOutput.contains(REJECT) || cleanedOutput.contains(REQUIRE) || cleanedOutput.contains(PREFER))) {
+                gradleRichVersions.putIfAbsent(dependencyGroupName, gavPieces.get(2));
+                richVersionGroup.computeIfAbsent(projectName, value -> new HashSet<>()).add(dependencyGroupName);
+            }
+
+            if(gradleRichVersions.containsKey(dependencyGroupName)) {
+                if(richVersionGroup.containsKey(projectName) && richVersionGroup.get(projectName).contains(dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                } else if(richVersionGroup.containsKey(rootProjectName) && richVersionGroup.get(rootProjectName).contains(dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                } else if (checkParentRichVersion(rootProjectName, projectParent, dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                }
+            }
+        }
+
         return gavPieces;
     }
 
+    private void addRelation(String projectParent, String projectName, String rootProjectName) {
+        if (!projectParent.equals("null") && !projectParent.contains("root project")) {
+            String parentString = projectParent.substring(projectParent.lastIndexOf(":") + 1, projectParent.lastIndexOf("'"));
+            relationsMap.putIfAbsent(projectName, parentString);
+        } else if (!projectParent.equals("null") && !projectName.equals(rootProjectName)) {
+            relationsMap.putIfAbsent(projectName, rootProjectName);
+        }
+    }
+
+    private boolean checkParentRichVersion(String rootProjectName, String projectParent, String dependencyGroupName) {
+       String currentProject = projectParent.substring(projectParent.lastIndexOf(":") + 1, projectParent.lastIndexOf("'"));
+       while(!currentProject.equals(rootProjectName)) {
+           if(richVersionGroup.containsKey(currentProject) && richVersionGroup.get(currentProject).contains(dependencyGroupName)) {
+               return true;
+           }
```

Comment:
Should we collect all 'parent' projects in rich version group before we exit out here or is it enough to exit when we find the nearest parent?

Reply:
Good point but there wont be anything to collect since every parent information would be present as we go over the dependency graphs on the basis of depth.